railties/test/application/rake/multi_dbs_test.rb

```diff
@@ -220,6 +302,32 @@ def db_test_prepare_name(name, schema_format)
         end
       end
 
+      def db_test_prepare_env(name, schema_format)
+        add_to_config "config.active_record.schema_format = :#{schema_format}"
+        require "#{app_path}/config/environment"
+
+        Dir.chdir(app_path) do
+          generate_models_for_animals
+
+          ENV["DATABASE_NAME"] = name
```

Comment:
Same comment here as above, this should be set in the tests not in the method. I've added an example to one of the tests below.